src/server/enterprise/server/api_server.go

```diff
@@ -287,3 +305,37 @@ func (a *apiServer) GetState(ctx context.Context, req *ec.GetStateRequest) (resp
 	}
 	return resp, nil
 }
+
+// Deactivate deletes the current cluster's enterprise token, and puts the
+// cluster in the "NONE" enterprise state. It also deletes all data in the
+// cluster, to avoid invalid cluster states. This call only makes sense for
+// testing
+func (a *apiServer) Deactivate(ctx context.Context, req *ec.DeactivateRequest) (resp *ec.DeactivateResponse, retErr error) {
+	a.LogReq(req)
+	defer func(start time.Time) { a.pachLogger.Log(req, resp, retErr, time.Since(start)) }(time.Now())
+
+	pachClient := a.getPachClient().WithCtx(ctx)
+	if err := pachClient.DeleteAll(); err != nil {
+		return nil, fmt.Errorf("could not delete all pachyderm data: %v", err)
+	}
+
+	if _, err := col.NewSTM(ctx, a.etcdClient, func(stm col.STM) error {
+		// blind delete
+		return a.enterpriseToken.ReadWrite(stm).Delete(enterpriseTokenKey)
+	}); err != nil {
+		return nil, err
+	}
+
+	// Wait until watcher observes the write
+	if err := backoff.Retry(func() error {
+		if t := a.enterpriseExpiration.Load().(time.Time); !t.IsZero() {
+			return fmt.Errorf("enterprise still activated")
+		}
+		return nil
+	}, backoff.RetryEvery(time.Second)); err != nil {
+		return nil, err
+	}
+	time.Sleep(time.Second) // give other pachd nodes time to observe the write
```

Comment:
Huh. I just read the code for the enterprise state. I see its cached, which makes sense ... you wouldn't want to issue an etcd query anytime the server needs to check for access. But that really only happens in the multi-pachd pod world, right? Couldn't you grab the lock / edit the value for this server, and that would be sufficient to update this pachd's enterprise state?

I'm not sure if this is something we need to change. In theory we do want to support multiple pachd pod deployments. On the other hand, this method is self described as a testing only method, and our tests currently only use a single pachd pod.

Reply:
Per our conversation: there are a lot of other things we could do here. The only reason there's any waiting at all is that tests were flaking, but we could:
1. update this server's cache, but still wait for other servers
2. update this server's cache, but don't wait for other servers
3. wait for this server's cache to update organically, but not wait for other servers

etc. We decided to leave this for now.